### PR TITLE
Dfe 502 updating charts for new api

### PR DESCRIPTION
**Pull request by negatron99**
Adding changes for DataBlock based on API changes

### CODE REVIEW

**Comment by markhiveit** [comment]
Build is failing -- I think it is specific to this branch:
`##[section]Starting: npm run build
==============================================================================
Task         : npm
Description  : Install and publish npm packages, or run an npm command. Supports npmjs.com and authenticated registries like Package Management.
Version      : 1.151.0
Author       : Microsoft Corporation
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613746)
==============================================================================
SYSTEMVSSCONNECTION exists true
SYSTEMVSSCONNECTION exists true
[command]C:\windows\system32\cmd.exe /D /S /C "C:\npm\prefix\npm.cmd --version"
6.9.0
[command]C:\windows\system32\cmd.exe /D /S /C "C:\npm\prefix\npm.cmd config list"
; cli configs
metrics-registry = "https://registry.npmjs.org/"
scope = ""
user-agent = "npm/6.9.0 node/v10.15.3 win32 x64"

; environment configs
cache = "C:\\npm\\cache"
prefix = "C:\\npm\\prefix"
userconfig = "d:\\a\\1\\npm\\9030.npmrc"

; builtin config undefined

; node bin location = C:\Program Files\nodejs\node.exe
; cwd = d:\a\1\s\src\explore-education-statistics-admin
; HOME = C:\Users\VssAdministrator
; "npm config ls -l" to show all defaults.

[command]C:\windows\system32\cmd.exe /D /S /C "C:\npm\prefix\npm.cmd run build"

> explore-education-statistics-admin@0.1.0 build d:\a\1\s\src\explore-education-statistics-admin
> node scripts/build.js

Creating an optimized production build...
Failed to compile.

d:/a/1/s/src/explore-education-statistics-admin/src/pages/prototypes/components/PrototypePublicationService.ts
TypeScript error: Type '{ subjectId: number; filters: number[]; indicators: number[]; startYear: number; endYear: number; geographicLevel: GeographicLevel.National; countries: never[]; localAuthorities: never[]; localAuthorityDistricts: never[]; regions: never[]; }' is not assignable to type 'DataBlockRequest'.
  Types of property 'startYear' are incompatible.
    Type 'number' is not assignable to type 'string'.  TS2322

    377 |         type: 'DataBlock',
    378 |         body: '',
  > 379 |         dataBlockRequest: {
        |         ^
    380 |           subjectId: 1,
    381 |           filters: [1, 2],
    382 |           indicators: [23, 26, 28],


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! explore-education-statistics-admin@0.1.0 build: `node scripts/build.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the explore-education-statistics-admin@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\npm\cache\_logs\2019-05-13T09_25_28_763Z-debug.log
Found npm debug log, make sure the path matches with the one in npm's output: C:\npm\cache\_logs\2019-05-13T09_25_28_763Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\npm\\prefix\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'build' ]
2 info using npm@6.9.0
3 info using node@v10.15.3
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle explore-education-statistics-admin@0.1.0~prebuild: explore-education-statistics-admin@0.1.0
6 info lifecycle explore-education-statistics-admin@0.1.0~build: explore-education-statistics-admin@0.1.0
7 verbose lifecycle explore-education-statistics-admin@0.1.0~build: unsafe-perm in lifecycle true
8 verbose lifecycle explore-education-statistics-admin@0.1.0~build: PATH: C:\npm\prefix\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;d:\a\1\s\src\explore-education-statistics-admin\node_modules\.bin;C:\agents\2.150.3\externals\git\cmd;C:/hostedtoolcache/windows/Python/3.6.8/x64;C:/hostedtoolcache/windows/Python/3.6.8/x64/Scripts;C:\cf-cli;C:\Program Files (x86)\NSIS\;C:\Program Files\Boost\1.69.0;C:\Users\VssAdministrator\.dotnet\tools;C:\Program Files\dotnet;C:\mysql-5.7.21-winx64\bin;C:\Program Files\Java\zulu-8-azure-jdk_8.36.0.1-8.0.202-win_x64\bin;C:\npm\prefix;C:\hostedtoolcache\windows\Ruby\2.5.0\x64\bin;C:\Go1.12.1\bin;C:\Program Files\Git\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\mingw64\bin;C:\hostedtoolcache\windows\Python\3.7.2\Scripts;C:\hostedtoolcache\windows\Python\3.7.2;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\ProgramData\Chocolatey\bin;C:\Program Files\Docker;C:\Program Files\PowerShell\6\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;c:\tools\php;C:\Program Files (x86)\Subversion\bin;C:\Program Files\nodejs\;C:\Program Files\CMake\bin;C:\Program Files\Mercurial;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\Program Files\OpenSSL\bin;C:\Users\VssAdministrator\AppData\Local\Microsoft\WindowsApps;
9 verbose lifecycle explore-education-statistics-admin@0.1.0~build: CWD: d:\a\1\s\src\explore-education-statistics-admin
10 silly lifecycle explore-education-statistics-admin@0.1.0~build: Args: [ '/d /s /c', 'node scripts/build.js' ]
11 silly lifecycle explore-education-statistics-admin@0.1.0~build: Returned: code: 1  signal: null
12 info lifecycle explore-education-statistics-admin@0.1.0~build: Failed to exec build script
13 verbose stack Error: explore-education-statistics-admin@0.1.0 build: `node scripts/build.js`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\npm\prefix\node_modules\npm\node_modules\npm-lifecycle\index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:189:13)
13 verbose stack     at ChildProcess.<anonymous> (C:\npm\prefix\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:189:13)
13 verbose stack     at maybeClose (internal/child_process.js:970:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
14 verbose pkgid explore-education-statistics-admin@0.1.0
15 verbose cwd d:\a\1\s\src\explore-education-statistics-admin
16 verbose Windows_NT 10.0.17763
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\npm\\prefix\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
18 verbose node v10.15.3
19 verbose npm  v6.9.0
20 error code ELIFECYCLE
21 error errno 1
22 error explore-education-statistics-admin@0.1.0 build: `node scripts/build.js`
22 error Exit status 1
23 error Failed at the explore-education-statistics-admin@0.1.0 build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

##[error]Error: Npm failed with return code: 1
##[section]Finishing: npm run build
`

File: src/explore-education-statistics-common/src/services/dataBlockService.ts [hunk 1]
```diff
@@ -124,30 +125,30 @@ interface ResponseMetaData {
 export interface DataBlockMetadata {
   indicators: ObjectMap<LabelValueUnitMetadata>;
   filters: ObjectMap<LabelValueMetadata>;
-  timePeriods: ObjectMap<LabelValueMetadata>;
+  timePeriods: ObjectMap<TimePeriod>;
```
Comment (ntsim) [inline src/explore-education-statistics-common/src/services/dataBlockService.ts]: We already have a util type for this called `Dictionary`

Comment (negatron99) [reply src/explore-education-statistics-common/src/services/dataBlockService.ts]: It's not quite the same though. I'm using "name" as the key. And it's a value coming from the API

Comment (ntsim) [reply src/explore-education-statistics-common/src/services/dataBlockService.ts]: They are the same type technically though, regardless of if they index name is called 'name' or 'key'. Would prefer we use the `Dictionary` type to be consistent here.

Comment (negatron99) [reply src/explore-education-statistics-common/src/services/dataBlockService.ts]: Done

**Pull request merged**